### PR TITLE
Feat: Add an overlay gradient

### PR DESCRIPTION
**Pull request by aronshamash**
- The sky banner and possibly any text overlaying part of an image has a gradient with transparency. This is not currently supported by the `gradient` package at the moment so this PR extends the package to add an `OverlayGradient` component.
- This will be needed for the sky sports banner in `video` as part of  work for https://nidigitalsolutions.jira.com/browse/REPLAT-3902

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/d4485482f3a80d5d54d597293af437ce3966763a/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/c95e23cb9fe6da6802f157935e9c001826e4eb98/index.html

File: packages/gradient/__tests__/web/__snapshots__/gradient.web.test.js.snap [hunk 1]
```diff
@@ -24,3 +28,5 @@ exports[`2. gradient with a child 1`] = `
   </div>
 </div>
 `;
+
+exports[`3. overlay gradient 1`] = `<div />`;
```
Comment (colincclark) [inline packages/gradient/__tests__/web/__snapshots__/gradient.web.test.js.snap]: this doesn't look right?

File: packages/gradient/gradient.showcase.js [hunk 2]
```diff
@@ -46,6 +47,37 @@ export default {
           }}
         />
       )
+    },
+    {
+      type: "story",
+      name: "Overlay gradient",
+      component: ({ number }) => (
+        <View>
```
Comment (colincclark) [inline packages/gradient/gradient.showcase.js]: Fragment?

File: packages/gradient/gradient.showcase.js [hunk 3]
```diff
@@ -46,6 +47,37 @@ export default {
...
+          <OverlayGradient
+            degrees={number("Gradient Angle: ", 180)}
```
Comment (colincclark) [inline packages/gradient/gradient.showcase.js]: should we allow the degrees / direction of the gradient to change at all by a consumer?

File: packages/gradient/gradient.showcase.js [hunk 4]
```diff
@@ -46,6 +47,37 @@ export default {
...
+              position: "absolute",
+              top: 0,
+              left: 0,
+              paddingTop: spacing(2),
+              paddingLeft: spacing(1),
+              width: 685,
+              zIndex: 3
+            }}
+          >
+            <Text style={{ color: "white" }}>Some example text</Text>
```
Comment (colincclark) [inline packages/gradient/gradient.showcase.js]: This `Text` component should be DRYed up and passed in a `color` as necessary

File: packages/gradient/src/gradient.base.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+const { LinearGradient, Path, Shape, Surface } = ART;
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1
+  },
+  surface: { position: "absolute", top: 0, left: 0, bottom: 0, right: 0 }
+});
+
+function angleToPoints(angle) {
```
Comment (colincclark) [inline packages/gradient/src/gradient.base.js]: this should go to a `utils` folder within this package/src

File: packages/gradient/src/gradient.base.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
+import React, { Component } from "react";
+import { ART, StyleSheet, View } from "react-native";
+import { defaultProps, propTypes } from "./gradient-prop-types.base";
+
+const { LinearGradient, Path, Shape, Surface } = ART;
+
+const styles = StyleSheet.create({
```
Comment (colincclark) [inline packages/gradient/src/gradient.base.js]: this should go to a `styles` fodler

File: packages/gradient/src/overlay-gradient.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { defaultProps, propTypes } from "./gradient-prop-types";
```
Comment (colincclark) [inline packages/gradient/src/overlay-gradient.js]: this file doesn't seem to exist?

File: packages/gradient/src/overlay-gradient.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import GradientBase from "./gradient.base";
+
+const OverlayGradient = ({ children, degrees, style }) => (
+  <GradientBase
+    degrees={degrees}
+    endColour="rgba(0, 0, 0, 0)"
+    startColour="rgba(1, 1, 1, .9)"
```
Comment (colincclark) [inline packages/gradient/src/overlay-gradient.js]: these should be named and put into the `colours.functional` object from `styleguide`

File: packages/gradient/gradient.showcase.js (hunk 3 above)
Comment (aronshamash) [reply packages/gradient/gradient.showcase.js]: Probably not butthis is the current API of gradient. Should be a new ticket to refactor?

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/45013d988bdd562f45df9d140b42606109644d0c/index.html

File: packages/gradient/src/overlay-gradient.js (hunk 7 above)
Comment (aronshamash) [reply packages/gradient/src/overlay-gradient.js]: should be there.

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/9242790b2b744d181c73dafb56eca6e8c49a35e1/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/1421cd8b8a8c26a7a9f787ec0e79b0e6b5b9e199/index.html

File: packages/gradient/gradient.showcase.js (hunk 3 above)
Comment (colincclark) [reply packages/gradient/gradient.showcase.js]: something that should have been picked up in a refinement

File: packages/gradient/src/gradient.js [hunk 9]
```diff
@@ -1,87 +1,23 @@
...
+  </GradientBase>
+);
 
 Gradient.propTypes = propTypes;
 Gradient.defaultProps = defaultProps;
 
 export default Gradient;
+export {
+  default as OverlayGradient
+} from "@times-components/gradient/src/overlay-gradient";
```
Comment (craigbilner) [inline packages/gradient/src/gradient.js]: this should be relative?

File: packages/gradient/__tests__/web/__snapshots__/gradient.web.test.js.snap [hunk 10]
```diff
@@ -24,3 +28,11 @@ exports[`2. gradient with a child 1`] = `
   </div>
 </div>
 `;
+
+exports[`3. overlay gradient with a child 1`] = `
```
Comment (craigbilner) [inline packages/gradient/__tests__/web/__snapshots__/gradient.web.test.js.snap]: does this provide value as a snapshot?

File: packages/gradient/gradient.showcase.js (hunk 3 above)
Comment (aronshamash) [reply packages/gradient/gradient.showcase.js]: yes agree.

File: packages/gradient/README.md [hunk 11]
```diff
@@ -16,24 +44,21 @@ Please see our main [README.md](../README.md) to get the project running locally
 ## Development
 
 The code can be formatted and linted in accordance with the agreed standards.
-
-```
-yarn fmt
-yarn lint
 ```
 
+yarn fmt yarn lint
```
Comment (colincclark) [inline packages/gradient/README.md]: bad formatting here - does not match any other README

File: packages/gradient/README.md [hunk 12]
```diff
@@ -16,24 +44,21 @@ Please see our main [README.md](../README.md) to get the project running locally
...
 platform with [jest](https://facebook.github.io/jest/).
-
-```
-yarn test:all
-yarn test:android
-yarn test:ios
-yarn test:web
 ```
 
+yarn test:all yarn test:android yarn test:ios yarn test:web
```
Comment (colincclark) [inline packages/gradient/README.md]: and here

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/db2dcd22b0e1f27111c58beb3bae0aaad72248a5/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/b73e6da9caa2e550abec9f3ea424b006b548981c/index.html

File: packages/gradient/src/gradient.base.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      height: 0,
+      width: 0
+    };
+  }
+
+  onLayout = ({ nativeEvent: { layout: { width, height } } }) => {
+    this.setState({ width, height });
```
Comment (colincclark) [inline packages/gradient/src/gradient.base.js]: code coverage does not cover this line

File: packages/gradient/src/overlay-gradient.js [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { colours } from "@times-components/styleguide";
+import { defaultProps, propTypes } from "./gradient-prop-types";
+import GradientBase from "./gradient.base";
+
+const OverlayGradient = ({ children, degrees, style }) => (
+  <GradientBase
```
Comment (colincclark) [inline packages/gradient/src/overlay-gradient.js]: code coverage does not cover this line in web

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/7f583b94f4db4136a95ec4085f99f6a26d48dd80/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/dcef31376850dd20132e49daf35fe1dbf3069b70/index.html

File: packages/gradient/README.md [hunk 15]
```diff
@@ -1,8 +1,37 @@
...
+exported components:
+
+* `Gradient` - used for image placeholders and loading screens
+* `OverlayGradient` - developed for use when overlaying text over parts of an
+  image
+
+## How to use
+
+```js
+import Gradient, { OverlayGradient } from "@times-components/gradient";
```
Comment (craigbilner) [inline packages/gradient/README.md]: `OverlayGradient` is unused?

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/c0f2601f9ff67b69d76c20d6efcf561af19fbb65/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/5bb112566a236540be998ed7c43474ef57ad406e/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/69f6cc7a97851fe99081c15a176679aa19932ca6/index.html

File: packages/gradient/README.md (hunk 15 above)
Comment (aronshamash) [reply packages/gradient/README.md]: added an example

File: packages/gradient/__tests__/web/overlay-gradient.web.test.js [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+
+      expect(
+        mount(<OverlayGradient degrees={30} style={styles.gradient} />)
+      ).toMatchSnapshot();
+    }
+  },
+  {
+    name: "overlay gradient with a child",
+    test() {
+      addSerializers(
```
Comment (craigbilner) [inline packages/gradient/__tests__/web/overlay-gradient.web.test.js]: I don't think adding serializers at this level works as intended and can be confusing, best to keep serializers at the file level

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/f959adf7d4aff074011d5466c290f97ea500ad89/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/5ab0beacd8fd31a4eed91b6af788ca8417374eb0/index.html

File: packages/gradient/src/gradient.base.js (hunk 13 above)
Comment (aronshamash) [reply packages/gradient/src/gradient.base.js]: added a test

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/605a8d21ee3fd27ee1056b0ac314f8e1f607ee21/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/90e85e0d2706901225ce2335cb3f58054f4e849a/index.html

**Pull request merged**